distributed/nanny.py

```diff
@@ -762,4 +762,6 @@ async def run():
             # Loop was stopped before wait_until_closed() returned, ignore
             pass
         except KeyboardInterrupt:
-            pass
+            # At this point the loop is not running thus we have to run
```

Comment:
Can you add a test for this?

Do you know if catching KeyboardInterrupt and doing some cleanup is preferred to registering a signal handler? https://www.tornadoweb.org/en/stable/ioloop.html#tornado.ioloop.IOLoop.add_callback_from_signal

Reply:
From the experience I had when trying to debug this before filing #3576 , I would definitely argue in favor of a simpler solution as the one proposed by this PR. Debugging all the async state in distributed that was causing this issue has proven really difficult and time consuming.

Reply:
Added a test.

In this case, I think it is more clear if we do it in the `except KeyboardInterrupt` catch because it is a specific handle of a `KeyboardInterrupt` while executing `loop.run_sync(run)`.

Reply:
It doesn't look like Tornado would catch [a `KeyboardInterrupt`]( https://github.com/tornadoweb/tornado/search?q=KeyboardInterrupt&unscoped_q=KeyboardInterrupt ) or [a `BaseException`]( https://github.com/tornadoweb/tornado/search?q=BaseException&unscoped_q=BaseException ). So suspect a Tornado-based signal handler won't work. Am I missing anything here?

It's probably possible to use Python's signal handler, but that may be more complicated than needed (especially since we are already catching `KeyboardInterrupt` here) and we will still have to restart the loop in any case.